File: src/main/java/com/zepben/vertxutils/routing/RouteGroup.java
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.zepben.vertxutils.routing;

import com.zepben.annotations.EverythingIsNonnullByDefault;

import java.util.List;

@EverythingIsNonnullByDefault
public interface RouteGroup {

    String mountPath();

    List<Route> routes();

    static RouteGroup create(String mountPath, List<Route> routes) {
        return new RouteGroup() {
            @Override
            public String mountPath() {
                return mountPath;
            }

            @Override
            public List<Route> routes() {
                return routes;
            }
        };
    }
}

File: src/main/java/com/zepben/vertxutils/routing/RouteGroup.kt
/*
 * Copyright 2026 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.vertxutils.routing

interface RouteGroup {

    val mountPath: String
    val routes: List<Route>

    companion object {

        fun create(mountPath: String, routes: List<Route>): RouteGroup =
            object : RouteGroup {
                override val mountPath: String = mountPath

                override val routes: List<Route> = routes
            }

    }

}
